### PR TITLE
feat(dropdowns): functional dropdown components and testing

### PR DESCRIPTION
**Pull request by alexsasharegan**

### CODE REVIEW

**Comment by alexsasharegan** [comment]
Moving the code coverage needle `0.67%` at a time 👍

**Comment by tmorehouse** [comment]
That might actually be a `.666%` rounded up to two decimals :p

File: lib/components/dropdown-header.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+export default {
+    functional: true,
+    props,
+    render(h, { props, data, children }) {
+        return h(
+            props.tag,
+            mergeData(data, {
+                staticClass: "dropdown-header",
+                attrs: { tabindex: "-1" }
```
Comment (tmorehouse) [inline lib/components/dropdown-header.js]: TabIndex no longer needed.

File: lib/components/index.js [hunk 2]
```diff
@@ -78,11 +78,17 @@ export {
     bCardGroup,
     bCardImg,
     bDropdown,
+    bDropdown as bDd,
     bDropdownItem,
+    bDropdownItem as bDdItem,
```
Comment (tmorehouse) [inline lib/components/index.js]: 👍

File: lib/components/dropdown-header.js (hunk 1 above)
Comment (tmorehouse) [inline lib/components/dropdown-header.js]: The main dropdown mixin doesn't focus headers any more.  users will need to point aria-describedby attribute to a  header with an ID.

Comment (alexsasharegan) [inline lib/components/dropdown-header.js]: Should I remove this then?

Comment (alexsasharegan) [inline lib/components/dropdown-header.js]: Oh, you did!

**Pull request merged**